### ISSUE TITLE
Change default language used for strings

### ISSUE THREAD

**Issue by SergioCardenas-Santex** [mac-cain13/R.swift#710]
Is there a more generic way to set the language used by strings?

I'm in a project that is recently adding i18n and was already using R.swift. I also know that we can do it so by replacing our strings
from _R.string.content.generatedName()_
to _R.string.content.generatedName(preferredLanguages: ["es"])_

The problem is we would need to change hundreds of strings manually. Instead, I would like to change **hostingBundle** or **applicationLocale**, which are both **fileprivate static let**. Could we have a method added to change one or both of those variables?

**Comment by skeyboy** [comment]
Prefer to add global settings for string language

**Comment by shadyk** [comment]
did u find a solution for the above ?

**Comment by mithilesh-parmar** [comment]
Any update on this?

**Comment by shadyk** [comment]
@SergioCardenas-Santex what did you do from your side ?

**Comment by mithilesh-parmar** [comment]
Here is my solution to this problem (Does not completely solve the mentioned problem but is a workaround): 

- Save user preferred language in UserDefaults
- Create an extension on Array with a computed variable returning the latest value of preferred language from UserDefaults
- pass the computed variable from extension to the param `preferredLanguages `  .

> extension Array where Element == String {
>
>    static var preferredLanguageCode: [String] { [ return [ // get language code from UserDefaults ] ] }
>
>}



Usage :

`R.string.localizable.continueTitle(preferredLanguages: .preferredLanguageCode`

You can refer to gist below:

https://gist.github.com/mithilesh-parmar/3eaa23dca0ac3be3fea701ab8f95eb3e.js

**Comment by shadyk** [comment]
@mithilesh-parmar what did you do from you side?

**Comment by mithilesh-parmar** [comment]
@shadyk  I changed all strings in my codebase. 

- Made an extension on Array 

> extension Array where Element == String {
>    static var preferredLanguageCode: [String] { [  // user preferred language code  ] }
>}

- Changed all strings to 

`R.string.localizable.someString(preferredLanguages: .preferredLanguageCode)`

**Comment by shadyk** [comment]
@mithilesh-parmar I was trying to escape this solution. Thanks anyway.

if the creators can change the **hosting bundle**'s access control to **public** it will be great

**Comment by mithilesh-parmar** [comment]
Yeah I know if there was a way to change hosting bundle it would have been great. But I had to ship my app  and could not find any solution so changed all the strings in codebase. @shadyk
